src/main/java/edu/harvard/iq/dataverse/License.java

```diff
@@ -86,20 +89,20 @@ public void setShortDescription(String shortDescription) {
         this.shortDescription = shortDescription;
     }
 
-    public URI getUri() {
-        return uri;
+    public URI getUri() throws URISyntaxException {
+        return new URI(uri);
     }
 
     public void setUri(URI uri) {
-        this.uri = uri;
+        this.uri = uri.toASCIIString();
     }
 
-    public URL getIconUrl() {
-        return iconUrl;
+    public URI getIconUrl() throws URISyntaxException {
```

Comment:
When would this throw a URISyntaxException? And how could this realistiscally happen?

Reply:
@janvanmansum This would throw a URISyntaxException if the URI String in the DB isn't a valid URI. Realistically this would never happen because a URI can only be inserted or updated in the DB if it's a correct URI object, I tested this. However, if someone directly tampers with the data in the DB, the exception might be thrown.

Reply:
As discussed, please wrap in an IllegalStateException. Rationale: the database has been corrupted in some way.